osu.Game/Overlays/BeatmapSet/Scores/ScoreTable.cs

```diff
@@ -167,12 +166,17 @@ private Drawable[] createContent(int index, ScoreInfo score)
 #pragma warning restore 618
             };
 
-            var availableStatistics = score.GetStatisticsForDisplay().ToDictionary(tuple => tuple.Result);
+            var availableStatistics = score.GetStatisticsForDisplay().ToLookup(touple => touple.DisplayName);
 
-            foreach (var result in statisticResultTypes)
+            foreach (var (columnName, resultTypes) in statisticResults)
             {
-                if (!availableStatistics.TryGetValue(result.result, out var stat))
-                    stat = new HitResultDisplayStatistic(result.result, 0, null, result.displayName);
+                HitResultDisplayStatistic stat = new HitResultDisplayStatistic(resultTypes.First(), 0, null, columnName);
+
+                if (availableStatistics.Contains(columnName))
+                {
+                    foreach (var s in availableStatistics[columnName])
+                        stat = s;
```

Comment:
I feel that this is dodgy because it only works on the assumption that if two hit results alias to the same display name, only one of them will be non-zero, while I believe other places would *sum up* the two values. See also: https://github.com/ppy/osu-web/pull/11062.

Reply:
In that case it should probably also be changed here (this is from the implementation you liked in the issue I used as a reference): https://github.com/ppy/osu/blob/790f863e0654fd563b57ab699d6be86895e756ab/osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCountController.cs#L46-L47
I'm not sure this is in scope here though

Reply:
No actually, you have this backwards. What you linked _already works the way I said above_.

`results` you linked there _is keyed by `HitResult`_. There is no coalescing of results yet at that stage. The coalescing of results is here:

https://github.com/ppy/osu/blob/790f863e0654fd563b57ab699d6be86895e756ab/osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCountController.cs#L56-L57
https://github.com/ppy/osu/blob/790f863e0654fd563b57ab699d6be86895e756ab/osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCountController.cs#L78-L87

On every hit judgement, the appropriate counter is retrieved from `results` and {in,de}cremented. _But that doesn't mean that two `HitResult`s can't point to the same counter._ And _if_ they do, that will have _the precise effect that I mentioned_ of _summing_ the counts.